minesweeper/src/Grid/Core.hs

```diff
@@ -0,0 +1,33 @@
+-- Define the Grid module for core functionality and rendering
+module Grid.Core
+    ( Grid
+    , renderGrid
+    ) where
+
+import Cell (Cell(..))
+
+-- Define the Grid type as a 2D list of Cell
+type Grid = [[Cell]]
```

Comment:
Using lists may be a tad slow. Can you explore which dimensions cause a noticeable lag?